apps/protocol-frontend/src/components/ContributionsTable.tsx

```diff
@@ -105,156 +103,107 @@ const ContributionsTable = ({
     });
   };
 
-  const data = useMemo<ContributionTableType[]>(() => {
-    const tableData = [] as ContributionTableType[];
-    for (const page of contributionsData) {
-      for (const contribution of page) {
-        tableData.push({
-          name: contribution.name,
-          txHash: contribution.tx_hash,
-          id: contribution.id,
-          details: contribution.details,
-          proof: contribution.proof,
-          updatedAt: contribution.updatedAt,
-          date_of_submission: contribution.date_of_submission,
-          engagementDate: formatDate(contribution.date_of_engagement),
-          date_of_engagement: formatDate(contribution.date_of_engagement),
-          attestations: contribution.attestations || null,
-          user: contribution.user,
-          activityTypeId: contribution.activity_type.id,
-          activity_type: contribution.activity_type,
-          guilds: contribution.guilds,
-          status: contribution.status,
-          action: '',
-          guildName:
-            contribution.guilds.map(guildObj => guildObj.guild.name)[0] ??
-            '---',
-        });
-      }
-    }
-    return tableData;
+  const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
+  const [selectedRows, setSelectedRows] = useState<UIContribution[]>([]);
+  const [sorting, setSorting] = useState<SortingState>([]);
+  const [globalFilter, setGlobalFilter] = useState('');
+
+  const data = useMemo<UIContribution[]>(() => {
```

Comment:
I think this might be where we lost the pagination support -- 

The nested array makes sure we account for the newly added page sets in the parent array